osu.Game/Screens/Backgrounds/BackgroundScreenDefault.cs

```diff
@@ -48,9 +59,57 @@ private void display(Background newBackground)
             currentDisplay++;
         }
 
+        private void onUserChanged(ValueChangedEvent<User> user)
+        {
+            if (BackgroundMode.Value == MainMenuBackgroundMode.Default)
+                return;
+
+            if (user.OldValue?.IsSupporter == user.NewValue?.IsSupporter)
+                return;
+
+            UpdateBackground();
+        }
+
+        private Texture skinBackground(Skin skin) => skin.GetTexture("menu-background");
+
+        private bool skinHasBackground => skin.Value.GetTexture("menu-background") != null;
+
+        private void onSkinChanged(ValueChangedEvent<Skin> skin)
+        {
+            if (BackgroundMode.Value != MainMenuBackgroundMode.Skin)
+                return;
+
+            if ((!User.Value?.IsSupporter) ?? true)
+                return;
+
+            if (skinBackground(skin.OldValue) == skinBackground(skin.NewValue))
+                return;
+
+            UpdateBackground();
+        }
+
+        private void onBeatmapChanged(ValueChangedEvent<WorkingBeatmap> beatmap)
+        {
+            if (BackgroundMode.Value == MainMenuBackgroundMode.Skin && skinHasBackground)
```

Comment:
I suggest to call `UpdateBackground` on some condition instead of performing `return`